Mac screensaver app can not be built on Travis CI

The Xcode ss_app target can not be built on Travis CI because of missing symbols. The error message seems to point to freetype or FTGL or some other missing static library. Since this is not release critical (it only happens on the Travis OSX VM) I just disabled the target but it would be nice if someone can investigate and fix it.

```
Undefined symbols for architecture x86_64:
  "_png_error", referenced from:
      _read_data_from_FT_Stream in libfreetype.a(sfnt.o)
  "_png_create_info_struct", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_set_read_fn", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_read_info", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_set_palette_to_rgb", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_get_error_ptr", referenced from:
      _error_callback in libfreetype.a(sfnt.o)
      _read_data_from_FT_Stream in libfreetype.a(sfnt.o)
  "_png_create_read_struct", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_set_longjmp_fn", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
      _error_callback in libfreetype.a(sfnt.o)
  "_png_set_tRNS_to_alpha", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_set_expand_gray_1_2_4_to_8", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_set_strip_16", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_read_image", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_set_packing", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_set_gray_to_rgb", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_set_interlace_handling", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_read_update_info", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_read_end", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_get_io_ptr", referenced from:
      _read_data_from_FT_Stream in libfreetype.a(sfnt.o)
  "_png_get_valid", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_get_IHDR", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_set_read_user_transform_fn", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_set_filler", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
  "_png_destroy_read_struct", referenced from:
      _Load_SBit_Png in libfreetype.a(sfnt.o)
```

--- a/mac_build/buildMacBOINC-CI.sh
+++ b/mac_build/buildMacBOINC-CI.sh
@@ -2,7 +2,7 @@
 
 # This file is part of BOINC.
 # http://boinc.berkeley.edu
-# Copyright (C) 2017 University of California
+# Copyright (C) 2022 University of California
 #
 # BOINC is free software; you can redistribute it and/or modify it
 # under the terms of the GNU Lesser General Public License
@@ -86,7 +86,6 @@ if [ ${share_paths} = "yes" ]; then
     fi
     source BuildMacBOINC.sh ${config} -all -setting HEADER_SEARCH_PATHS "../clientgui ${cache_dir}/include ../samples/jpeglib ${cache_dir}/include/freetype2" -setting USER_HEADER_SEARCH_PATHS "" -setting LIBRARY_SEARCH_PATHS "$libSearchPathDbg ${cache_dir}/lib ../lib" | tee xcodebuild_all.log | $beautifier; retval=${PIPESTATUS[0]}
     if [ $retval -ne 0 ]; then
-        curl --upload-file ./xcodebuild_all.log https://transfer.sh/xcodebuild_all.log
         cd ..; exit 1; fi
     return 0
 fi
@@ -109,121 +108,152 @@ if [ "${style}" == "Development" ]; then
     libSearchPathDbg="${cache_dir}/lib/debug"
 fi
 target="mgr_boinc"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} -setting HEADER_SEARCH_PATHS "../clientgui ${cache_dir}/include" -setting LIBRARY_SEARCH_PATHS "${libSearchPathDbg} ${cache_dir}/lib" | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 ## Target gfx2libboinc also build dependent target jpeg
 target="gfx2libboinc"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} -setting HEADER_SEARCH_PATHS "../samples/jpeglib" | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 target="libboinc"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 target="api_libboinc"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 target="PostInstall"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 target="switcher"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 target="gfx_switcher"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 target="Install_BOINC"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
-# screensaver disabled because Travis can't build some library correctly, see https://github.com/BOINC/boinc/issues/2662
-#libSearchPath="./build/Deployment"
-#if [ "${style}" == "Development" ]; then
-#    libSearchPath="./build/Development"
-#fi
-#target="ss_app"
-#source BuildMacBOINC.sh ${config} -noclean -target ${target} -setting HEADER_SEARCH_PATHS "../api/ ../samples/jpeglib/ ${cache_dir}/include ${cache_dir}/include/freetype2"  -setting LIBRARY_SEARCH_PATHS "${libSearchPath} ${cache_dir}/lib" | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
-#if [ ${retval} -ne 0 ]; then
-#    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
-#    cd ..; exit 1;
-#fi
+libSearchPath=""
+if [ "${style}" == "Development" ]; then
+   libSearchPath="./build/Development"
+fi
+target="ss_app"
+echo "Building ${target}..."
+source BuildMacBOINC.sh ${config} -noclean -target ${target} -setting HEADER_SEARCH_PATHS "../api/ ../samples/jpeglib/ ${cache_dir}/include ${cache_dir}/include/freetype2"  -setting LIBRARY_SEARCH_PATHS "${libSearchPath} ${cache_dir}/lib ../lib" | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
+if [ ${retval} -ne 0 ]; then
+    echo "Building ${target}...failed"
+    cd ..; exit 1;
+fi
+echo "Building ${target}...done"
 
 target="ScreenSaver"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} -setting GCC_ENABLE_OBJC_GC "unsupported" | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 target="boinc_opencl"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 target="setprojectgrp"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 target="cmd_boinc"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 target="Uninstaller"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 target="SetUpSecurity"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 target="AddRemoveUser"
+echo "Building ${target}..."
 source BuildMacBOINC.sh ${config} -noclean -target ${target} | tee xcodebuild_${target}.log | $beautifier; retval=${PIPESTATUS[0]}
 if [ ${retval} -ne 0 ]; then
-    curl --upload-file ./xcodebuild_${target}.log https://transfer.sh/xcodebuild_${target}.log
+    echo "Building ${target}...failed"
     cd ..; exit 1;
 fi
+echo "Building ${target}...done"
 
 cd ..

